prov-api/routes/model_run/model_run.py

```diff
@@ -164,3 +168,71 @@ async def register_model_run_sync(
             success=True, details=f"Provenance record lodged successfully."),
         record_info=result
     )
+
+
+@router.post(f"{edit_root_prefix}/link_to_study", response_model=AddStudyLinkResponse, operation_id="add_study_link")
+async def link_to_study(
+    model_run_id: str,
+    study_id: str,
+    roles: ProtectedRole = Depends(read_write_user_protected_role_dependency),
+    config: Config = Depends(get_settings)
+) -> AddStudyLinkResponse:
+
+    # validate model_run_id and study_id to be linked.
+    request_style = RequestStyle(
+        user_direct=roles.user, service_account=None)
+    
+    model_run, study = await validate_model_run_study_linking(
+        model_run_id=model_run_id,
+        study_id=study_id,
+        request_style=request_style,
+        config=config
+    )
+    model_run: ItemModelRun
```

Comment:
what is this doing?

Reply:
![image](https://github.com/user-attachments/assets/1499a6f5-90da-40fb-bbe0-72866deb23fa)
for some reason my mypy typing wasnt working properly so just added that simple line to tell it what it is i think its allg